### DISCUSSION TITLE
"Markdown" messages?

### DISCUSSION THREAD

**Discussion by nic0lette** [Skylar-Tech/node-red-contrib-matrix-chat#89] [Q&A] [answered]
Is it possible to send messages using the markdown flavor supported by Matrix?

That is, something like: `> This is quoted text`, or `This text is a **bold** message`?

**Comment by skylord123** [comment, marked as answer]
I created this issue to track this:
https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/issues/53

So currently the front-end of element converts markdown to HTML and HTML is all the matrix server sees. We would have to port the part of code from matrix-react-sdk that does this conversion.

In the mean time you can manually do HTML. What I do is send a message formatted in markdown then view the message source in element to see how it converts to HTML then build your flows using the HTML equivalent.

If you need a list of supported HTML tags you can use that can be found here:
https://github.com/matrix-org/matrix-react-sdk/blob/a1328d8ef70c6dba620caeab56b3c0a6867f79b4/src/HtmlUtils.tsx#L234-L297

And for future reference this is the part of matrix-react-sdk that parses the markdown to HTML:
https://github.com/matrix-org/matrix-react-sdk/blob/a1328d8ef70c6dba620caeab56b3c0a6867f79b4/src/editor/serialize.ts#L141

But yeah, this would definitely be handy to have built-in.

**Comment by nic0lette** [comment]
Oh, awesome. And TIL that this is done in the front-end client, not the server =o Thank you!
